dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/statelessMode/StatelessModeTest.java

```diff
@@ -0,0 +1,344 @@
+import org.skyscreamer.jsonassert.JSONAssert;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+import componenttest.topology.utils.HttpRequest;
+import io.openliberty.mcp.internal.fat.tool.basicToolApp.BasicTools;
+import io.openliberty.mcp.internal.fat.tool.cancellationApp.CancellationTools;
+
+/**
+ *
+ */
+@RunWith(FATRunner.class)
+public class StatelessModeTest extends FATServletClient {
+
+    private static final String ACCEPT_HEADER = "application/json, text/event-stream";
+    private static final String MCP_PROTOCOL_HEADER = "MCP-Protocol-Version";
+    private static final String MCP_PROTOCOL_VERSION = "2025-06-18";
+
+    @Server("mcp-stateless-server")
+    public static LibertyServer server;
+    private static ExecutorService executor;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "statelessModeTest.war")
+                                   .addPackage(BasicTools.class.getPackage())
+                                   .addPackage(CancellationTools.class.getPackage());
+
+        ShrinkHelper.exportDropinAppToServer(server, war, SERVER_ONLY);
+
+        server.startServer();
+
+        executor = Executors.newSingleThreadExecutor();
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
+        server.stopServer();
+    }
+
+    @AfterClass
+    public static void shutdownExecutor() {
+        executor.shutdown();
+    }
+
+    private static final String ENDPOINT = "/statelessModeTest/mcp";
+
+    @Test
+    public void testInitializeDoesNotReturnSessionIdInStatelessMode() throws Exception {
+        String request = """
+                        {
+                          "jsonrpc": "2.0",
+                          "id": 1,
+                          "method": "tools/call",
+                          "params": {
+                            "name": "echo",
+                            "arguments": {
+                              "input": "Hello"
+                            }
+                          }
+                        }
+                        """;
+
+        HttpRequest httpRequest = new HttpRequest(server, ENDPOINT)
+                                                                   .requestProp("Accept", ACCEPT_HEADER)
+                                                                   .requestProp(MCP_PROTOCOL_HEADER, MCP_PROTOCOL_VERSION)
+                                                                   .jsonBody(request)
+                                                                   .method("POST");
+
+        String response = httpRequest.run(String.class);
+        assertFalse("Expected no session ID in stateless mode (in body)", response.contains("sessionId"));
```

Comment:
session ID would be in the header: https://modelcontextprotocol.io/specification/2025-06-18/basic/transports#session-management